components/src/button/button.directive.ts

```diff
@@ -74,8 +58,14 @@ export class ButtonComponent implements OnInit, OnChanges {
       "focus:tw-ring",
       "focus:tw-ring-offset-2",
       "focus:tw-ring-primary-700",
-      this.block ? "tw-w-full tw-block" : "tw-inline-block",
-      buttonStyles[this.buttonType ?? "secondary"],
-    ];
+    ]
+      .concat(this.block ? ["tw-w-full", "tw-block"] : ["tw-inline-block"])
+      .concat(buttonStyles[this.buttonType ?? "secondary"]);
```

Comment:
We're doubling up on setting the default value here - `buttonType` already has a default value, I don't think you need the null coalescing in this line.